corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/api.js

```diff
@@ -47,7 +47,16 @@ FormplayerFrontend.module("Menus", function (Menus, FormplayerFrontend, Backbone
                                 response.exception || FormplayerFrontend.Constants.GENERIC_ERROR,
                                 response.type === 'html'
                             );
-                            FormplayerFrontend.trigger('navigation:back');
+
+                            var currentUrl = FormplayerFrontend.getCurrentRoute();
+                            if (FormplayerFrontend.lastError === currentUrl) {
+                                FormplayerFrontend.lastError = null;
+                                FormplayerFrontend.trigger('navigateHome');
+                            } else {
+                                FormplayerFrontend.lastError = currentUrl;
```

Comment:
There is probably a better way of storing this information rather than in this variable on the global object. I couldn't immediately find anything else where shared scope between requests was stored (other than the URL).
Any other suggestions on how to implement this?

Reply:
How much work would it be to store the history of navigation changes as a global object? That would allow us to detect this and might be useful state to keep around? I'm not sure how complicated it would be though but it loosk like it's all in the router https://github.com/dimagi/commcare-hq/blob/5ca2ffe38023b0412c906d1aac1830bf0e4fa032/corehq/apps/cloudcare/static/cloudcare/js/formplayer/router.js#L123?

Reply:
Yeah, I actually looked at what it would take to store the whole thread of where you've gone in a separate object, but then realized that this already exists to some extent (`History`), and that having this list accessible was a bit of YAGNI since I really only needed to know whether a specific route had an error state or not. 

AFAIK `History` doesn't allow you to examine its contents as it is seen as a security hole.

Reply:
Storing this on the Global seems to me like it is reasonably inline with the architecture of the marionette page.